assignments/array-methods.js

```diff
@@ -55,29 +56,128 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
-console.log(fullName);
+
+// let fullName = [];
+// for (let i = 0; i < runners.length; i++) {
+//     fullName.push(runners[i].first_name + " " + runners[i].last_name)
+// }
+// console.log(fullName)
+
+
+let fullName2 = []
+runners.forEach(runner =>  fullName2.push(`${runner.first_name} ${runner.last_name}`))
+console.log(fullName2);
+
+
+let fullName4 = []
+runners.forEach(addName)
+
+function addName(word){
+    fullName4.push(`${word.first_name} ${word.last_name}`)
+}
+
+console.log(fullName4)
+
+
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+
+// let allCaps = [];
+let allCaps = runners.map(allCapMode)
+
+function allCapMode (words) {
+    return words.first_name.toUpperCase()
+    // allCaps.push(`${wubba.first_name}`)
+}
+
+
+
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. 
+//Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+
+
+let largeShirts = runners.filter(shirts).map(names)
+
+function shirts(words){
+if (words.shirt_size === "L") {
+    return true
+}
+else {
+    return false
+}
+}
```

Comment:
Why not use the `.includes()` method here? It returns a true if the argument passed to it is included in the array, or srting it was called on, and false if it does not.